src/jit/codegencommon.cpp

```diff
@@ -10204,7 +10204,19 @@ var_types Compiler::GetHfaType(CORINFO_CLASS_HANDLE hClass)
     {
 #ifdef FEATURE_HFA
         CorInfoType corType = info.compCompHnd->getHFAType(hClass);
-        if (corType != CORINFO_TYPE_UNDEF)
+#ifdef _TARGET_ARM64_
+        if (corType == CORINFO_TYPE_VALUECLASS)
+        {
+            // This is a vector type.
+            // HVAs are only supported on ARM64, and only for sizes of 8 or 16 bytes.
+            // For 8-byte vectors corType will be returned as CORINFO_TYPE_DOUBLE.
```

Comment:
This isn't classified independently as a short vector?

Reply:
No; for ABI purposes it is the same (an 8-byte vector is passed in the lower 8-bytes of the register, just as for double)

Reply:
Is this something that might be useful to differentiate at some point in the future?

Reply:
It may be, but it will come at a (probably not large) price in complexity and change to the JIT/EE interface (unless the need for it is internal to the JIT), so I'd propose we wait until such time as we need it.